core/state/snapshot/snapshot.go

```diff
@@ -117,6 +117,11 @@ var (
 	ErrNotConstructed = errors.New("snapshot is not constructed")
 )
 
+type Opt struct {
+	BlockHash  common.Hash
+	ParentHash common.Hash
+}
```

Comment:
not sure where this should live or if it should have a different structure. would also be used for triedb update